### DISCUSSION TITLE
How to do "select * from(subQuery)"?

### DISCUSSION THREAD

**Discussion by hellhoundisalab** [slick/slick#2863] [Questions]
I have a complex subquery (written in slick), that has a distinct on primary key tuple in it: `select distinct on (intId, strId) ... from table where ...`. I also need to sort the resulting set using arbitrary columns/set of columns, so to accomplish this i would write sth like `select * from(select distinct on (intId, strId) ... from table where ...) order by ...`. How do I do that in slick? Were it not for the composite primary key I would write `table.filter(_.id in subQuery).sortBy(...)`. I don't want to include a sorted column in a depends on clause because that might tank performance quite a lot.

**Comment by nafg** [comment]
I don't fully understand the question, but does adding `.subquery` help? Like `table.filter(_.id in subQuery).subquery.sortBy(...)`.
